gateway/src/main/kotlin/Gateway.kt

```diff
@@ -141,6 +188,48 @@ public suspend inline fun Gateway.start(token: String, config: GatewayConfigurat
     start(builder.build())
 }
 
+/**
+ * Starts a reconnecting gateway connection with the given parameters.
+ *
+ * After connecting, a resume request will be sent and the session will be resumed if the session is valid.
+ * If the session was invalidated, an identify request will be sent and a session will be created.
+ * If you want to start a new session, use [start].
+ *
+ * This function will suspend until the lifecycle of the gateway has ended.
+ *
+ * ```kotlin
+ * gateway.resume("your_token", "gatewaySessionId", "https://discord-resume-url-here.discord.com/", 0) {
+ *     shard = DiscordShard(0,1)
+ *
+ *     presence {
+ *         afk = false
+ *         status = Status.Online
+ *         watching("you :eyes:")
+ *     }
+ *
+ * }
+ *
+ * //gateway has disconnected
+ * ```
+ *
+ * @param token     The Discord token of the bot.
+ * @param session   The gateway session information.
+ * @param config additional configuration for the gateway.
+ */
+public suspend inline fun Gateway.resume(token: String, session: GatewaySession, config: GatewayConfigurationBuilder.() -> Unit = {}) {
```

Comment:
Why do we need this function?

Reply:
If you want to resume a session with a different gateway configuration, example: You want to change the bot's presence for the next time the bot reidentifies.

Reply:
Did not think about this 👍